arviz/stats/stats.py

```diff
@@ -633,36 +635,22 @@ def summary(data, var_names=None, include_circ=None, stat_funcs=None,
         ...
         >>> az.summary(trace, ['mu'], stat_funcs=[trace_sd, trace_quantiles])
                  sd     5    50    95
-        mu__0  0.06  0.00  0.10  0.21
-        mu__1  0.07 -0.16 -0.04  0.06
+        mu[0]  0.06  0.00  0.10  0.21
+        mu[1]  0.07 -0.16 -0.04  0.06
     """
     posterior = convert_to_dataset(data, group="posterior")
-
-    if var_names is None:
-        var_names = list(posterior.data_vars)
+    posterior = posterior if var_names is None else posterior[var_names]
 
     if batches is None:
         batches = min([100, posterior.draw.size])
 
-    alpha = 1 - credible_interval
-
-    def make_stat_ufunc(func, **kwargs):
-        def circ_ufunc(ary):
-            return func(ary.reshape(*ary.shape[:-2], -1), **kwargs, axis=-1)
-
-        return circ_ufunc
-
-    def make_mc_error_ufunc(**kwargs):
-        def mc_error_ufunc(ary):
-            return _mc_error(ary.reshape(*ary.shape[:-2], -1).T, **kwargs)
-
-        return mc_error_ufunc
-
-    def make_hpd_ufunc(idx, **kwargs):
-        def hpd_ufunc(ary):
-            return hpd(ary.reshape(*ary.shape[:-2], -1).T, **kwargs)[..., idx]
+    fmt_group = ("wide", "long", "xarray")
+    if not isinstance(fmt, str) or (fmt not in fmt_group):
```

Comment:
I think checking `str` object type here may be redundant here because of the check for `in fmt_group`

Reply:
`.lower` for non str

Reply:
I don't quite understand. Is `.lower()` a method on non strings?

Reply:
Oh I see, yes this makes sense, you're avoiding the exception that would be raised if a non string was passed